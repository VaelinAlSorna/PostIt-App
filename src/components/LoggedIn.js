import React from 'react';
import { Link } from 'react-router-dom';

const LoggedIn = (props) => (
    <div>
        <h3>Logged In As:</h3>
        <p>{props.name}</p>
        <Link to="/create"><button>Create Article</button></Link>
        <button onClick={props.handleFilter}>{props.filter ? 'See All Articles' : 'See Your Articles'}</button>
        <button onClick={props.onLogout}>Logout</button>
    </div>
);

export default LoggedIn;
