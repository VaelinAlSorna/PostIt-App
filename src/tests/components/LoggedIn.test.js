import React from 'react';
import { shallow } from 'enzyme';
import LoggedIn from '../../components/LoggedIn';
import user from '../fixtures/usersFixtures';

test('should redner LoggedIn correctly', () => {
    // I am not testin when filter is false, but I don't think thats necessary
    const wrapper = shallow(<LoggedIn 
                                name={user.name}
                                filter={true}
                                handleFilter={jest.fn()}
                                onLogout={jest.fn()}   
                            />)
    expect(wrapper).toMatchSnapshot();
});