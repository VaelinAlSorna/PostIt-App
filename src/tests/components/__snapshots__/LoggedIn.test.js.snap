// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should redner LoggedIn correctly 1`] = `
<div>
  <h3>
    Logged In As:
  </h3>
  <p>
    Mark
  </p>
  <Link
    replace={false}
    to="/create"
  >
    <button>
      Create Article
    </button>
  </Link>
  <button
    onClick={[Function]}
  >
    See All Articles
  </button>
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</div>
`;
